### PR TITLE
Language-specific documentation for C and C++

### PR DESCRIPTION
**Pull request by ResistantBear**
# Language-specific documentation for C and C++

Closes: #239

## Description

- Added and linked a language-specific documentation file for C++
- Added and linked a language-specific documentation file for C
- Added some missing test cases for C and C++

## Definition of Done

A PR is only ready for merge once all the following acceptance criteria are fulfilled:
* [x]  All TODOs related to this PR have been closed
* [x]  There are automated tests for newly written code and bug fixes
* [ ]  All bugs discovered while working on this PR have been submitted as issues (if not already an open issue)
* [x]  Documentation ([README.md](https://github.com/MaibornWolff/metric-gardener/blob/main/README.md), [UPDATE_GRAMMARS.md](https://github.com/MaibornWolff/metric-gardener/blob/main/UPDATE_GRAMMARS.md)) has been updated

### CODE REVIEW

File: C.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# File Metrics for C
```
Comment (ce-bo) [inline C.md:1]: I think it could be clearer to put all these markdown files into a separate docs folder.

Comment (ResistantBear) [reply C.md:1]: @mylinhdao @clemens-mw I have seen on another PR that you have already done such a thing and are disussing the naming of the folder still. Can one of you to take care of moving the `C.md` and `CPP.md` file to your new subfolder and adjusting the links to them in the `README.md` when merging? Thank you.

Comment (clemens-mw) [reply C.md:1]: Yes, see #255

File: cpp_example_code.o
Comment (clemens-mw) [inline cpp_example_code.o:1]: delete this

**Review by clemens-mw — CHANGES_REQUESTED** [review]
@mylinhdao also added some tests and a fix for `xor` in her pull request #256 
should we open a new issue or integrate those here?

File: CPP.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+-   Any function calls, including calls of standard library functions for a more functional programming style, like `std::for_each`
+-   Any preprocessor directives, including `#if`, `#ifdef` and `#elif`
+-   Bitwise binary operators, like `^` (`xor`), `&` (`bitand`), `|` (`bitor`), `&=` (`and_eq`) or `|=` (`or_eq`)
+
+### functions
+
+The "functions" metric counts function declarations in both header and source files. This also includes the definition of functions in source or header files that were already declared before. The metric is tested to account for:
+
+-   general functions
+-   member functions, including constructors, (pure) virtual functions, default functions and deleted functions
```
Comment (mylinhdao) [inline CPP.md:33]: should we mention that destructor is also counted?

**Review by mylinhdao — APPROVED** [review]
minor remark :)

**Review by clemens-mw — APPROVED** [review]

File: CPP.md (hunk 2 above)
Comment (ResistantBear) [reply CPP.md]: Fixed

**Pull request merged**